### DISCUSSION TITLE
Univariate Timeseries - which models can be used with ts_learner on prediction on 7 days horizon

### DISCUSSION THREAD

**Discussion by qwertz747** [timeseriesAI/tsai#356] [Q&A] [answered]
Hi,

thx again for this great work.
I am having a problem understanding the functionality of **ts_learner**.
Working with an univariate time series and created the X,y, and split with the following code. So I have a lookbackwindow of 4 weeks and I try to forecast 1 week ahead. 

`window_length = 7*4 #length of lookback window
stride = 1 
horizon=7 
seq_first=True`
`X, y = SlidingWindow(window_length, stride=stride, horizon=horizon, seq_first=seq_first, sort_by=average_kg.index, ascending=True)(average_kg)`

The x,y,splits have the form:
![image](https://user-images.githubusercontent.com/36340314/148694314-f2db3f1a-83ec-4603-93ab-3def52315fe1.png)

When I try to build the learner it seems to work only for certain models. 
Is there something I am missing, what models can be used for univariate forecasting/prediction regarding a horizon of a week?

For example:
**Does not work:**
- learn = ts_learner(dls, **LSTM**, metrics=[mae, rmse], cbs=ShowGraph())
TypeError: __init__() got an unexpected keyword argument 'custom_head'

- learn = ts_learner(dls, **LSTM_FCN**, metrics=[mae, rmse], cbs=ShowGraph())
TypeError: __init__() got an unexpected keyword argument 'custom_head'

- learn = ts_learner(dls, **MiniRocketPlus**, metrics=[mae, rmse], cbs=ShowGraph())
TypeError: __init__() got an unexpected keyword argument 'custom_head'

- learn = ts_learner(dls, **InceptionTime**, metrics=[mae, rmse], cbs=ShowGraph())
TypeError: __init__() got an unexpected keyword argument 'custom_head'

**Works:**
LSTMPlus
GRUPlus

I tried all the same models with TSForecaster too and the same models work or don't work: 
`batch_tfms = TSStandardize()
fcst = TSForecaster(X, y, splits=splits, batch_tfms=batch_tfms, bs=512, arch=GRUPlus, metrics=rmse, cbs=ShowGraph())
fcst.fit_one_cycle(50, 1e-1)
fcst.export("models/fcst.pkl")`

Can someone please point me in the right direction how to forecast a week ahead with more models (MiniRocketPlus & LSTM_FCN would be great)?

**Comment by oguiza** [comment, marked as answer]
Hi @qwertz747,
Sorry for my late response.
All models ending in Plus (except MiniRocketPlus which is a special type of model) should work. The Plus actually indicates that additional properties have been added to the original model.

**Reply by qwertz747** [reply]
Thx for your reply/info!
